### PR TITLE
test application submit & log in/out

### LINKED ISSUE #29: sign in/out
write tests to ensure sign in/out button works properly

### LINKED ISSUE #30: add tests for applicationq
application submits properly when you click submit

### PR DESCRIPTION
**Pull request by NovaSagittarii**

### CODE REVIEW

**Comment by NovaSagittarii** [comment]
oops its a bit messy in terms of commits but it works
i was working with more network stubs

**Comment by NovaSagittarii** [comment]
i got bonked by prettier now checking e2e
and the package.json isnt updated :frowning:

**Comment by NovaSagittarii** [comment]
aaaa i realized so many problems are because i forgot to pull dev before i started working on it... 

well if its too scuffed i can try reimplement from scratch so git history is cleaner

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -37,29 +37,36 @@
...
+  // We need to refresh this cookie once in a while.
+  // We are unsure if this is true and if true, when it needs to be refreshed.
+  cy.setCookie('next-auth.session-token', Cypress.env('SESSION_TOKEN'));
+  cy.setCookie(
+    '__Secure-next-auth.session-token',
+    Cypress.env('SESSION_TOKEN'),
+    { secure: true }
+  );
+});
+Cypress.Commands.add('mock', () => {
```
Comment (minsooerickim) [inline cypress/support/commands.ts]:
```suggestion
Cypress.Commands.add('mockAPI', () => {
```

can you rename this to mockAPI and make according changes in the test functions

File: cypress/e2e/sign-in-out.cy.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+describe('Check sign out button (sign in is mocked)', () => {
+  it('make sure log out succeeds', () => {
+    // setup spying before request happens
+    cy.server();
```
Comment (minsooerickim) [inline cypress/e2e/sign-in-out.cy.ts]: i think `.server()` is a deprecated function? I don't think you even need this line to run this function correctly

**Review by minsooerickim — CHANGES_REQUESTED** [review]
noice job as always 💯

File: cypress/support/commands.ts (hunk 1 above)
Comment (NovaSagittarii) [reply cypress/support/commands.ts]: oops bad variable naming , will fix :+1:

File: cypress/e2e/sign-in-out.cy.ts (hunk 2 above)
Comment (NovaSagittarii) [reply cypress/e2e/sign-in-out.cy.ts]: oh, i thought i removed the other cy.server()'s when i was templating the intercept methods and then removed it after i found out it does nothing. thanks for catching this. 🙏

**Comment by minsooerickim** [comment]
@NovaSagittarii squashed a few commits, everything looks good 💯

**Pull request merged**